### PR TITLE
📨 Add email capture backend

### LINKED ISSUE #7: 📨 Email capture backend
## Summary
Create the email backend that intercepts outgoing emails instead of sending them.

## Requirements
- Compatible with standard Python email sending patterns
- Captures email message and stores via EmailStore
- Works with fastapi-mail, aiosmtplib patterns
- No actual SMTP connection

## Acceptance Criteria
- [ ] `mailview/backend.py` with capture backend
- [ ] Async `send` method that stores instead of sends
- [ ] Parse email.message.Message objects
- [ ] Extract HTML, plaintext, attachments
- [ ] Unit tests for various email formats

## Dependencies
- #5 Email data model
- #6 SQLite store layer

### PR DESCRIPTION
**Pull request by swmcc**
## Summary
- Implements `MailviewBackend` class that intercepts outgoing emails instead of sending
- Parses `email.message.Message` and `EmailMessage` objects from Python stdlib
- Extracts all email components: sender, recipients (to/cc/bcc), subject, HTML body, text body, headers, attachments
- Stores captured emails via `EmailStore`
- Supports sender/recipient overrides for integration flexibility
- Includes `capture_email()` convenience function for one-off captures
- Exports new components from package root

## Test plan
- [x] 20 unit tests covering parsing and capture
- [x] Tests for simple text, HTML, multipart emails
- [x] Tests for multiple recipients
- [x] Tests for attachments (single and multiple)
- [x] Tests for header extraction
- [x] Tests for sender/recipient overrides
- [x] Edge case tests (empty recipients, no subject, no body)

Closes #7

---
Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: mailview/backend.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        await self.store.save(email)
+        return email
+
+    def parse_message(
+        self,
+        message: Message | EmailMessage,
+        *,
+        sender: str | None = None,
+        recipients: Sequence[str] | None = None,
+    ) -> Email:
```
[the remark this answers is not in the document]
Comment (swmcc) [reply mailview/backend.py]: Added - new TestLegacyMessage class with tests for parsing legacy Message objects via email.message_from_string(), including both simple and multipart messages.
